bin/oneoff/migrate_applab_to_firebase

```diff
@@ -0,0 +1,170 @@
+    raise "Failed to PUT data at #{url} with error #{response.code} '#{response.body}'"
+  end
+end
+
+def post_with_auth(url, data)
+  uri = URI.parse("#{url}?auth=#{CDO.firebase_secret}")
+  request = Net::HTTP::Post.new(uri)
+  request.body = data
+
+  response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") do |http|
+    http.request(request)
+  end
+
+  unless response.is_a?(Net::HTTPSuccess)
+    raise "Failed to POST data at #{url} with error #{response.code} '#{response.body}'"
+  end
+end
+
+# Sets the counters, metadata, and contents of a table in firebase.
+#
+# @param encrypted_channel_id [String]
+# @param properties_data [Hash] Map from property name to JSON-encoded value
+def set_firebase_properties(encrypted_channel_id, properties_data)
+  channel_url = "https://#{CDO.firebase_name}.firebaseio.com/v3/channels/#{encrypted_channel_id}"
+  properties_url = "#{channel_url}/storage/keys.json"
+  put_with_auth(properties_url, properties_data.to_json)
+end
+
+# Sets the counters, metadata, and contents of a table in firebase.
+#
+# @param encrypted_channel_id [String] An encrypted, base64-encoded channel id.
+# @param table_name [String]
+# @param table_records [Hash] Map from record id to json-encoded record
+# @params table_columns [Array] Array of column names
+def set_firebase_table(encrypted_channel_id, table_name, table_records, table_columns)
+  channel_url = "https://#{CDO.firebase_name}.firebaseio.com/v3/channels/#{encrypted_channel_id}"
+
+  storage_url = "#{channel_url}/storage/tables/#{table_name}/records.json"
+  storage_data = table_records.to_json
+  put_with_auth(storage_url, storage_data)
+
+  counters_url = "#{channel_url}/counters/tables/#{table_name}.json"
+  counters_data = {
+    lastId: table_records.keys.map(&:to_i).max,
+    rowCount: table_records.size
+  }.to_json
+  put_with_auth(counters_url, counters_data)
+
+  if table_columns
+    metadata_url = "#{channel_url}/metadata/tables/#{table_name}/columns.json"
+    put_with_auth(metadata_url, 'null')
+    table_columns = ['id'] + (table_columns - ['id'])
+    table_columns.each do |column_name|
+      post_with_auth(metadata_url, {columnName: column_name}.to_json)
+    end
+  end
+end
+
+# http://localhost-studio.code.org:3000/projects/applab/Ezx7frRzCh_QoTetQwzVZQ/edit
+# http://localhost.code.org:3000/v3/edit-csp-app/Ezx7frRzCh_QoTetQwzVZQ
+def stub_applab_channels
+  [
+    [1, 4307], # "Remix: My Project" (dynamodb deleted)
+    [1, 4385], # Ezx7frRzCh_QoTetQwzVZQ dynamodb with data
+    [1, 4386], # EosN50MleCkJgP-52bxCEQ dynamodb no data
+    [1, 4388], # 5Rep_hVVFfkCQR283rnQVg firebase with data
+    [1, 4389] # 6f5LdZdKnraOtKhxulEj5A firebase no data
+  ]
+end
+
+# Returns an array of pairs of storage_owner_ids and channel_ids, each representing an applab app.
+# There are ~1.1M of these on production.
+def fetch_applab_channels
+  PEGASUS_DB[:storage_apps].grep(:value, '%applab%').map do |row|
```

Comment:
How often does the reporting DB get updated? Regardless, using the reporting DB to get the initial list of applab projects sounds fine since no new DynamoDB projects are being created (all new applab projects use Firebase).

  however I'd be concerned about data consistency issues